docs/src/components/ComponentDoc/ComponentPropsTable/ComponentPropsRow.tsx

```diff
@@ -1,68 +1,76 @@
+import { ComponentProp, ComponentPropType } from 'docs/src/types'
 import ComponentPropName from '../ComponentProp/ComponentPropName'
 
-export default class ComponentPropsRow extends React.Component<any, any> {
-  static propTypes = {
-    defaultValue: PropTypes.string,
-    description: PropTypes.arrayOf(PropTypes.string),
-    name: PropTypes.string,
-    required: PropTypes.bool,
-    tags: PropTypes.array,
-    type: PropTypes.string,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
-  }
+const InlineMarkdown = React.lazy(() => import('../InlineMarkdown'))
+
+type ComponentPropsRowProps = ComponentProp
 
-  state: any = {}
+const ComponentPropValue: React.FunctionComponent<ComponentPropType> = props => {
+  const { name, parameters } = props
 
-  toggleEnums = () => this.setState({ showEnums: !this.state.showEnums })
+  if (!name) return <span>enum</span>
 
-  render() {
-    const { defaultValue, description, name, required, tags, type } = this.props
-    const hideRow = this.docSiteHidden(tags)
-    const slot = this.isSlotProp(type, tags)
-    const rowStyle: React.CSSProperties = {
-      borderTopWidth: '1px',
-      borderTopStyle: 'solid',
-      borderTopColor: 'grey',
-    }
+  if (name === 'ShorthandValue' || name === 'ShorthandCollection') {
+    const componentName = parameters[0].name.replace('Props', '')
 
-    // TODO: use Flex or a Table component, when it will be available
     return (
-      <tr className={hideRow ? 'hidden' : ''} style={rowStyle}>
-        <td>
-          <ComponentPropName name={name} required={required} slot={slot} />
-        </td>
-        <td>
-          <ComponentPropDefaultValue value={defaultValue} />
-        </td>
-        <td>{`{${type}}`}</td>
-        <td>
-          <ComponentPropDescription description={description} />
-          {/* TODO change these according to the react-docgen-typescript generated json */}
-          {/* <ComponentPropFunctionSignature name={name} tags={tags} /> */}
-          {/* <ComponentPropEnum */}
-          {/* showAll={showEnums} */}
-          {/* toggle={this.toggleEnums} */}
-          {/* type={type} */}
-          {/* values={value} */}
-          {/* /> */}
-        </td>
-      </tr>
+      <span>
+        {name}
+        {`<`}
+        <Link to={`/components/${componentName}`}>{parameters[0].name}</Link>
+        {`>`}
+      </span>
     )
   }
 
-  docSiteHidden(tags) {
-    return _.some(tags, ['title', 'docSiteIgnore'])
-  }
-  isSlotProp(type: string, tags): boolean {
-    if (type.startsWith('ShorthandValue') || type.startsWith('ShorthandCollection')) {
-      return true
-    }
-    return _.some(tags, ['title', 'slot'])
-  }
+  return <span>{name}</span>
+}
+
+const ComponentPropsRow: React.FunctionComponent<ComponentPropsRowProps> = props => {
+  const { defaultValue, description, name, required, types } = props
+
+  const shorthand = types.find(
```

Comment:
```suggestion
  const shorthand = types.some(
```

and `!!shorthand` -> `shorthand`